step3/dataStores/Storage.js

```diff
@@ -0,0 +1,48 @@
+const uuidv4 = require('uuid/v4');
+
+class Storage {
+
+	constructor () {
+		this.chat = {
+			messages: [],
+			users: []
+		}
+	}
+
+	saveMessage(sender, body) {
+		this.chat.messages.push({
+			sender: sender,
+			body: body,
+			timestamp: Date.now(),
+			id: uuidv4()});
+	}
+
+	getMessages() {
+		return this.chat.messages;
+	}
+
+	connect(username) {
+		const index = this.chat.users.indexOf(username);
+		if (index === -1) {
```

Comment:
`else`?

Reply:
Uważasz, że jest potrzebne? To jest używane w controllerze w taki sposob `		const connected = storage.connect(req.body.username);
		if (connected) {`
poprostu jak nic nie zwracamy to mamy `undefined` i warunek w if nie jest spełniony